pkg/provisioner/resourcehandlers.go

```diff
@@ -226,40 +226,36 @@ func deleteObjectBucket(ob *v1alpha1.ObjectBucket, c versioned.Interface) error
 	return nil
 }
 
-func updateClaim(c versioned.Interface, obc *v1alpha1.ObjectBucketClaim, retryInterval, retryTimeout time.Duration) (*v1alpha1.ObjectBucketClaim, error) {
-	err := wait.PollImmediate(retryInterval, retryTimeout, func() (done bool, err error) {
+func updateClaim(c versioned.Interface, obc *v1alpha1.ObjectBucketClaim, retryInterval, retryTimeout time.Duration) (result *v1alpha1.ObjectBucketClaim, err error) {
```

Comment:
named return `result` is unused, so a nil obc is always returned.

Reply:
good catch. A c&p error on my part. The code still works due to pointers being used but it is a bug.